src/bin/ion/commands/beta/count.rs

```diff
@@ -1,39 +1,42 @@
+use crate::commands::{IonCliCommand, WithIonCliArgument};
 use anyhow::{Context, Result};
-use clap::{Arg, ArgAction, ArgMatches, Command};
+use clap::{ArgMatches, Command};
 use ion_rs::*;
 use std::fs::File;
 use std::io::{stdin, BufReader, StdinLock};
 
-pub fn app() -> Command {
-    Command::new("count")
-        .about("Prints the number of top-level values found in the input stream.")
-        .arg(
-            // All argv entries after the program name (argv[0])
-            // and any `clap`-managed options are considered input files.
-            Arg::new("input")
-                .index(1)
-                .help("Input file [default: STDIN]")
-                .action(ArgAction::Append)
-                .trailing_var_arg(true),
-        )
-}
+pub struct CountCommand;
+
+impl IonCliCommand for CountCommand {
```

Comment:
🗺️ The `IonCliCommand` trait (and its doc comments) can be found in `commands/mod.rs`, which unfortunately appears quite late in this diff.